paderbox/array/sparse.py

```diff
@@ -397,16 +408,51 @@ def from_array_and_onset(cls, array, onset, shape):
         array([1., 1., 0., 0., 0., 0., 0., 0., 0., 0.])
         >>> SparseArray.from_array_and_onset(np.ones(5), 7, 10)
         SparseArray(_SparseSegment(onset=7, array=array([1., 1., 1.])), shape=(10,))
+        >>> SparseArray.from_array_and_onset(np.ones(5), 7).as_contiguous()
+        array([0., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1.])
         """
+        if shape is None:
+            shape = list(array.shape)
+            shape[0] += onset
+            if shape[0] < 0:
+                shape[0] = 0
         out = SparseArray(shape=shape)
-        if onset < 0:
-            array = array[..., -onset:]
-            onset = 0
-        if array.shape[-1] + onset > out.shape[-1]:
-            array = array[..., :out.shape[-1] - onset]
-        out._add_segment(_SparseSegment(onset, array))
+        out._add_segment(_SparseSegment(onset, array).get_clipped(out.shape[-1]))
```

Comment:
Should we disable, that _SparseSegment could represent a segment, that is partially outside?

Reply:
Added a method  `_SparseSegment.from_array` that constructs a clipped segment. This should be used everywhere where a `_SparseSegment` is constructed from an array that is not controlled by a SparseArray